include/parsed_mapped_functions.h

```diff
@@ -0,0 +1,181 @@
+    const std::string &parsed_component_names,
+    const std::string &parsed_id_components,
+    const std::string &parsed_id_functions,
+    const std::string &parsed_constants):
+  ParameterAcceptor(parsed_name),
+  name (parsed_name)
+{
+  component_names = Utilities::split_string_list(parsed_component_names, ',');
+  split_id_components(parsed_id_components);
+  split_id_functions(parsed_id_functions,parsed_constants);
+
+  AssertDimension(id_components.size(), id_functions.size());
+
+  AssertDimension(id_defined_components.size(), id_defined_functions.size());
+
+  Assert((std::is_permutation(id_defined_components.begin(),
+                              id_defined_components.end(),
+                              id_defined_functions.begin())),
+         ExcMessage("Ids associated to components and to functions are not the same."));
+
+  typedef std::map<unsigned int, ComponentMask>::iterator it_type;
+  for (it_type it=id_components.begin(); it != id_components.end(); ++it)
+    {
+      std::pair<ComponentMask, shared_ptr<ParsedFunction<spacedim,n_components> > > mapped(it->second, id_functions[it->first]);
+
+      mapped_functions[it->first] = mapped;
+
+    }
+
+};
+
+template <int spacedim, int n_components>
+void ParsedMappedFunctions<spacedim,n_components>::split_id_components(const std::string &parsed_id_components)
+{
+  std::vector<std::string> idcomponents;
+
+  idcomponents = Utilities::split_string_list(parsed_id_components, ',');
+
+  for (unsigned int i=0; i<idcomponents.size(); ++i)
+    {
+      std::vector<std::string> id_comp;
+      std::vector<std::string> components;
+      std::vector<bool> mask(n_components,false);
+
+      id_comp = Utilities::split_string_list(idcomponents[i], ':');
+
+      unsigned int id = Utilities::string_to_int(id_comp[0]);
+      id_defined_components.push_back(id);
+
+      components = Utilities::split_string_list(id_comp[1], ';');
+      for (unsigned int c=0; c<components.size(); ++c)
+        {
+          unsigned int m = Utilities::string_to_int(components[c]);
+          mask[m] = true;
+        }
+      id_components[id] = ComponentMask(mask);
+    }
+}
+
+
+template <int spacedim, int n_components>
+void ParsedMappedFunctions<spacedim,n_components>::split_id_functions(const std::string &parsed_id_functions,
+    const std::string &constants)
+{
+  std::vector<std::string> idfunctions;
+
+  idfunctions = Utilities::split_string_list(parsed_id_functions, ',');
+
+  for (unsigned int i=0; i<idfunctions.size(); ++i)
+    {
+      std::vector<std::string> id_func;
+
+      id_func = Utilities::split_string_list(idfunctions[i], ':');
+
+      unsigned int id = Utilities::string_to_int(id_func[0]);
+      id_defined_functions.push_back(id);
+
+      std::string function_name = name + " acting on id " + Utilities::int_to_string(id);
+      shared_ptr<ParsedFunction<spacedim,n_components> > pf;
+      pf = SP(new ParsedFunction<spacedim,n_components>(function_name, id_func[1], constants));
```

Comment:
This won't work, for the reason I said above. At construction time, you don't know what the name of the function will be. And if you do, then you can only use this with the name specified in the constructor. Use the trick I explained above.